Initialized Vitepress w/ Basic Information

Africk-Hunter opened the pull request:
I initialized Vitepress in the project & set up the file structure similarly to how it was in the old docs. Additionally, I have set up the markdown files that were already present in the repo to be used on the site under 'CS2KZ Gamemode Info.'

Africk-Hunter committed:
Vitepress Initialization

- Initialized Vitepress & Reorganized file structure to resemble the structure of the old docs website.

Africk-Hunter committed:
Removed Vitepress Default Information Files

Africk-Hunter committed:
Readability & Reformatting updates

- Rewrote filters section since it is very important but was previously a little confusing to properly grasp the concept of.
- Created a page for points and point formula
- Separated Map approval process into a different page as its more related to development than gameplay
- Rewrote or reformatted various other sections for better readability

Sikarii commented on .vitepress/config.mts:
```diff
@@ -0,0 +1,48 @@
+
+export default defineConfig({
+  title: "CS2 KZ Docs",
+  description: "CS 2 Kreedz documentation",
```
```suggestion
  description: "CS2 Kreedz documentation",
```

Sikarii commented on .vitepress/config.mts:
```diff
@@ -0,0 +1,48 @@
+function sidebar() {
+  return [
+    {
+      text: "CS2KZ Gamemode Info",
```
I think "CS2" should not be repeated in every section, and I also think these need a better name.

```suggestion
      text: "Systems",
```

Sikarii commented on .vitepress/config.mts:
```diff
@@ -0,0 +1,48 @@
+    {
+      text: "CS2KZ Gamemode Info",
+      items: [
+        { text: "Modes", link: "/guides/Modes" },
+        { text: "Maps", link: "/guides/Maps" },
+        { text: "Points", link: "/guides/Points" },
```
The files (and thus links) should probably prefer to not be case sensitive and instead be kebab-case.
```suggestion
        { text: "Modes", link: "/systems/modes" },
        { text: "Maps", link: "/systems/maps" },
        { text: "Points", link: "/systems/points" },
```

Sikarii commented on .vitepress/config.mts:
```diff
@@ -0,0 +1,48 @@
+      ],
+    },
+    {
+      text: "CS2KZ Creator Resources",
+      items: [
+        { text: "Map Approval Process", link: "/guides/MapApproval" },
```
I think mapping resources, as theres more to come (such as the Mapping API), should probably be under their own path.

```suggestion
      text: "Creator resources",
      items: [
        { text: "Map Approval Process", link: "/mapping/approval" },
```

Africk-Hunter committed:
Added Styles section + Updated README

- Added a separate section about styles since I didn't feel like it fit very well with the other sections.
- Updated README to provide more useful information
- Made minor writing changes to existing sections
- Removed Vitepress cache files (forgot to add them to gitignore, oops lol)

Sikarii commented on README.md:
```diff
@@ -1,9 +1,11 @@
 
-- [Modes](./Modes.md)
-- [Maps](./Maps.md)
+- [Modes](./docs/guides/Modes.md)
+- [Maps](./docs/guides/Maps.md)
```
Per above suggestions.

```suggestion
- [Modes](/docs/systems/modes)
- [Maps](/docs/systems/maps)
```

Sikarii commented on docs/index.md:
```diff
@@ -0,0 +1,31 @@
+that test their precision, speed, and movement skills. Compete against yourself or others, improve your abilities, 
+and climb the ranks as you master each course.
+
+## CS2KZ Gamemode Information
+
+- [Modes](/guides/Modes)
+- [Maps](/guides/Maps)
+- [Points](/guides/Points)
```
Per earlier suggestion about a better name.

```suggestion
## Systems

- [Modes](/systems/modes)
- [Maps](/systems/maps)
- [Points](/systems/points)
```

Sikarii commented on docs/guides/Maps.md:
```diff
@@ -0,0 +1,91 @@
+ defined by a start and an end zone. Unlike GOKZ, CS2KZ eliminates the distinction between
+  "main" and "bonus" courses—there are simply courses.
+
+To have a [map approved](./MapApproval.md), mappers must meet a minimum quality standard 
```
```suggestion
To have a [map approved](/mapping/approval), mappers must meet a minimum quality standard 
```

Sikarii commented on docs/guides/Maps.md:
```diff
@@ -0,0 +1,91 @@
+
+### Key Components of a Filter
+
+1. **Mode** - This defines how the course is played (either [CKZ](./Modes.md#Classic) or [VNL](./Modes.md#Vanilla))
```
```suggestion
1. **Mode** - This defines how the course is played (either [CKZ](/systems/modes#Classic) or [VNL](/systems/modes#Vanilla))
```

Sikarii commented on docs/guides/Points.md:
```diff
@@ -0,0 +1,45 @@
+
+### 1. Filter Points
+
+These points are tied to a specific [Filter](./Maps.md#filters) applied to a course and measure a player’s performance on that particular combination of map, mode, and settings.
```
```suggestion
These points are tied to a specific [Filter](/systems/maps#filters) applied to a course and measure a player’s performance on that particular combination of map, mode, and settings.
```

Sikarii commented on docs/index.md:
```diff
@@ -0,0 +1,31 @@
+- [Maps](/guides/Maps)
+- [Points](/guides/Points)
+
+## CS2KZ Creator Resources
+
+- [Map approval process](/guides/MapApproval)
```
```suggestion
## Creator resources

- [Map approval process](/mapping/approval)
```

Sikarii commented on index.md:
This file should not exist.

Sikarii commented on package.json:
```diff
@@ -0,0 +1,13 @@
+    "vitepress": "^1.5.0"
+  },
+  "scripts": {
+    "docs:dev": "vitepress dev",
+    "docs:build": "vitepress build",
+    "docs:preview": "vitepress preview"
```
Remove docs prefix as the repository is only docs, this will help later down the line with StackBlitz for example.

```suggestion
    "dev": "vitepress dev",
    "build": "vitepress build",
    "preview": "vitepress preview"
```

Sikarii commented on .vitepress/config.mts:
```diff
@@ -0,0 +1,48 @@
+  description: "CS 2 Kreedz documentation",
+  head: [['link', { rel: 'icon', href: '/public/favicon.ico' }]],
+  srcDir: "docs",
+
```
Enable [VitePress clean urls](https://vitepress.dev/reference/site-config#cleanurls) as GitHub Pages natively supports it.
Will allow for urls to be valid without the `.html` suffix.

```suggestion

cleanUrls: true
```

Sikarii reviewed (changes requested):
Hey, thanks a lot for putting in the work for all this.
I pointed out a few places where some ironing out might be needed.

I am still not sure if `Systems` is an appropriate name for the section, so if you do find a better one, feel free to go for it. Additionally, the filenames need to be adjusted if changes are made to the sub-paths.

Africk-Hunter committed:
Apply suggestions from code review

Co-authored-by: Sikari <37634135+Sikarii@users.noreply.github.com>

Africk-Hunter committed:
Applied review changes

- Recreated file structure to include mapping & systems folders

Sikarii commented on index.md:
This file should still not exist.

Africk-Hunter replied:
I'm a little confused on this, when I removed it, it kept giving me a 404 error on the root page.

Sikarii commented on docs/systems/maps.md:
```diff
@@ -0,0 +1,91 @@
+- Instead of a "main" course and 0 or more "bonus" courses we will just have
+  "courses".
+- Each course will have 4 "filters" applied to it. Each filter consists of a
+  [mode](./Modes.md), whether teleports are allowed, a tier, and a "ranked" status.
```
```suggestion
  [mode](/systems/modes), whether teleports are allowed, a tier, and a "ranked" status.
```

Sikarii reviewed (changes requested):
Awesome work, just a little things that were missed.
Should be good to go after these!

Africk-Hunter committed:
Delete index.md

Africk-Hunter committed:
Update docs/systems/maps.md

Co-authored-by: Sikari <37634135+Sikarii@users.noreply.github.com>

Sikarii reviewed (approved):
LGTM!
